File: psets-02/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    // Prompt user for text
    string text = get_string("Text: ");

    int letters = 0;
    int words = 0;
    int sentences = 0;

    int length = strlen(text);

    // Count letters, words, sentences
    for (int i = 0; i < length; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
        else if (text[i] == ' ')
        {
            words++;
        }
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }

    // Words = spaces + 1 (if text isn't empty)
    words++;

    // Calculate L and S
    float L = ((float) letters / words) * 100;
    float S = ((float) sentences / words) * 100;

    // Coleman-Liau index
    float index = 0.0588 * L - 0.296 * S - 15.8;

    int grade = round(index);

    // Output result
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

File: psets-06/readability.py
from cs50 import get_string

# Get text from user
text = get_string("Text: ")

letters = 0
words = 0
sentences = 0

# Count letters, words, sentences
for char in text:
    if char.isalpha():
        letters += 1
    elif char in [".", "!", "?"]:
        sentences += 1

# Words are separated by spaces
words = len(text.split())

# Compute L and S
L = (letters / words) * 100
S = (sentences / words) * 100

# Coleman-Liau index
index = round(0.0588 * L - 0.296 * S - 15.8)

# Output result
if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")
